### PR TITLE
Grtlv 77 ukea csat frontend fixes

### PR DESCRIPTION
**Pull request by FoxJamie16**
Many QA updates and fixes for the HCSAT frontend including content updates, styling, error scenarios and messages, required fields and an additional 'cancel' button for js and non-js scenarios (QA listed in ticket) . Form progression bugfixes also included.  

_Tick or delete as appropriate:_

### Workflow

- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/GRTLV-77
- [x] Jira ticket has the correct status.
- [x] A clear/description pull request messaged added.

### Reviewing help

- [x] Includes screenshot(s) - ideally before and after, but at least after

### Housekeeping

- [x] Frontend assets have been re-compiled.
- [x] I have checked that my PR is using the latest package versions of: great-components, directory-constants, directory-healthcheck, directory-validators, directory-components, directory-api-client, directory-ch-client, django-staff-sso-client, directory-forms-api-client, directory-sso-api-client, sigauth

### Merging

- [x] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)
Screenshot 2024-06-05 at 15 22 17
screencapture-greatcms-trade-great-8020-export-academy-registration-booking-4d169c40-b5b4-453b-bea6-cdc48af85926-success-2024-06-05-15_22_56
screencapture-greatcms-trade-great-8020-export-academy-registration-booking-4d169c40-b5b4-453b-bea6-cdc48af85926-success-2024-06-05-15_22_25

### CODE REVIEW

File: domestic/static/javascript/hcsat-feedback-form.js [hunk 1]
```diff
@@ -19,6 +22,11 @@ class CsatFormHandler {
             const formData = new FormData(this.form);
             formData.append('step', this.currentStep);
             const url = this.form.action
+            
+            if (event.submitter && event.submitter.name=='cancelButton'){
```
Comment (lewis-coulson-dit) [inline domestic/static/javascript/hcsat-feedback-form.js:26]: `event.submitter.name == 'cancelButton'`

File: domestic/static/javascript/hcsat-feedback-form.js [hunk 2]
```diff
@@ -19,6 +22,11 @@ class CsatFormHandler {
...
+                this.resetForm();
+                return
```
Comment (lewis-coulson-dit) [inline domestic/static/javascript/hcsat-feedback-form.js:28]: shouldn't need this return

Comment (FoxJamie16) [reply domestic/static/javascript/hcsat-feedback-form.js:28]: If this isn't included then the rest of the function fires as if the form was submitted, showing additional elements along with sending a request to the backend. Keeping the return prevents this

Comment (lewis-coulson-dit) [reply domestic/static/javascript/hcsat-feedback-form.js:28]: if / else ?

Comment (FoxJamie16) [reply domestic/static/javascript/hcsat-feedback-form.js]: Replaced with if/else, thanks @lewis-coulson-dit

File: domestic/static/javascript/hcsat-feedback-form.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+const e = require("venv/lib/python3.9/site-packages/wagtail/admin/static/wagtailadmin/js/vendor/jquery.datetimepicker");
```
Comment (lewis-coulson-dit) [inline domestic/static/javascript/hcsat-feedback-form.js:1]: whats this?

**Review by lewis-coulson-dit — APPROVED** [review]

File: domestic/static/javascript/hcsat-feedback-form.js (hunk 1 above)
Comment (boxadesign) [reply domestic/static/javascript/hcsat-feedback-form.js]: Safer to test for event.submitter before doing a name test imo.

File: domestic/static/javascript/hcsat-feedback-form.js (hunk 3 above)
Comment (boxadesign) [reply domestic/static/javascript/hcsat-feedback-form.js]: No idea!

File: domestic/static/javascript/hcsat-feedback-form.js (hunk 2 above)
Comment (boxadesign) [reply domestic/static/javascript/hcsat-feedback-form.js]: I think the return is ok here. Whats the issue with it?

**Pull request merged**